### ISSUE TITLE
SUGGESTION: Make phpservermon to use standardized plugins

### ISSUE THREAD

**Issue by raggoo** [phpservermon/phpservermon#595]
I see a lot of "have to rewrite" too much code.

I am not a coder so be gentle.

Could you not make phpservermon where it will accept standardized plugins for it and then others could just write plugins to add whatever and you wouldn't have to recode a bunch of stuff to accommodate it?

I see lots of things with plugins to add stuff to it so it only makes sense to me.

Sorry if any of my terminologies are wrong.

Thanks.

**Comment by TimZ99** [comment]
You're reading my mind!

The integration of Telegram required changes in 25 files (translation files not included), that's way too much.

Before the update, adding an SMS gateway required changes to a lot of files.
Now, after the update, adding an SMS gateway only needs 3 row added to 1 file and the rest of the implementation goes into a newly created file.

This is the first step to a more modular version of PSM

**Comment by mooleshacat** [comment]
Agreed... Make the plugin SDK simple too :+1:

**Comment by maltegrosse** [comment]
agree, needed to change 18 files for webhook support [webhook support](https://github.com/phpservermon/phpservermon/pull/951)
